pkg/ccl/backupccl/restore_span_covering.go

```diff
@@ -247,3 +249,75 @@ rangeLoop:
 	}
 	return requestEntries, maxEndTime, nil
 }
+
+// makeSimpleImportSpans partitions the spans of requiredSpans into a covering
+// of RestoreSpanEntry's which each have all overlapping files from the passed
+// backups assigned to them. The spans of requiredSpans are trimmed/removed
+// based on the lowWaterMark.
+func makeSimpleImportSpans(
+	requiredSpans roachpb.Spans,
+	backups []BackupManifest,
+	backupLocalityMap map[int]storeByLocalityKV,
+	lowWaterMark roachpb.Key,
+) []execinfrapb.RestoreSpanEntry {
+	if len(backups) < 1 {
+		return nil
+	}
+
+	for i := range backups {
+		sort.Sort(BackupFileDescriptors(backups[i].Files))
+	}
+
+	var cover []execinfrapb.RestoreSpanEntry
+	for _, span := range requiredSpans {
+		if span.EndKey.Compare(lowWaterMark) < 0 {
+			continue
+		} else if span.Key.Compare(lowWaterMark) < 0 {
+			span.Key = lowWaterMark
+		}
+
+		spanCoverStart := len(cover)
+
+		for layer := range backups {
+			covPos := spanCoverStart
+			// TODO(dt): binary search to the first file in required span?
+			for _, f := range backups[layer].Files {
+				if sp := span.Intersect(f.Span); sp.Valid() {
+					fileSpec := execinfrapb.RestoreFileSpec{Path: f.Path, Dir: backups[layer].Dir}
+					if dir, ok := backupLocalityMap[layer][f.LocalityKV]; ok {
+						fileSpec = execinfrapb.RestoreFileSpec{Path: f.Path, Dir: dir}
+					}
+					if len(cover) == spanCoverStart {
+						cover = append(cover, makeEntry(span.Key, sp.EndKey, fileSpec))
+					} else {
+						// Add each file to every matching partition in the cover.
```

Comment:
do you think this code could benefit from a ascii-art comment -- like the one on `frontier.SpanEntries`?

Reply:
ugh, yes.

I started one when I first wrote this and then removed it since I was spending more time trying to fix it up than the code it felt like, but I guess a little more investment now will save re-figuring it out later.

Added.